modules/mcagar/src/main/ts/ephox/mcagar/api/TinyHooks.ts

```diff
@@ -0,0 +1,124 @@
+import { after, afterEach, before } from '@ephox/bedrock-client';
+import { Arr, Fun, Optional, Strings, Type } from '@ephox/katamari';
+import { Insert, Remove, SugarBody, SugarElement, SugarShadowDom } from '@ephox/sugar';
+import 'tinymce';
+import { Editor as EditorType } from '../alien/EditorTypes';
+import * as Loader from '../loader/Loader';
+import { setTinymceBaseUrl } from '../loader/Urls';
+
+export interface Hook<T extends EditorType> {
+  readonly editor: () => T;
+}
+
+export interface ShadowRootHook<T extends EditorType> extends Hook<T> {
+  readonly shadowRoot: () => SugarElement<ShadowRoot>;
+}
+
+const setupBaseUrl = (tinymce: any, settings: Record<string, any>): void => {
+  if (settings.base_url) {
```

Comment:
Can we make this an explicit check?

Reply:
This was copied almost verbatim from `Loader`, so I'll update that as well.

Reply:
I've moved this function (to remove the duplication) and added a strict check.